lib/Command/ExpirePassword.php

```diff
@@ -106,6 +116,19 @@ protected function execute(InputInterface $input, OutputInterface $output) {
 			$value
 		);
 
+		// add a dummy password in the user_password_history so the cron job
+		// can notify about the expiration of the password. We'll use "dummy"
+		// as password hash (we won't hash "dummy")
+		$latestPassword = $this->mapper->getLatestPassword($uid);
+		if ($latestPassword->getPassword() !== 'dummy') {
+			// insert dummy only if it isn't the last
+			$oldPassword = new OldPassword();
+			$oldPassword->setUid($uid);
+			$oldPassword->setPassword('dummy');
+			$oldPassword->setChangeTime(\max($date->getTimestamp(), $latestPassword->getChangeTime() + 1));
```

Comment:
The notification would not work if we set a negative time with the occ command as the `max` will always take the timestamp of the latest password. I think this is ok as a negative value would trigger immediate expiry, and immediate expiry notification. This is ok.

If an admin runs the occ command twice with two different values, you must still insert a new "dummy" entry to have the most recent timestamp value. This would also keep track indirectly in history table about how many times an admin has changed the expiration time.

Reply:
- [ ] doubt about `$date->getTimestamp()`. if through occ we say "password must expire in 5 days", we shouldn't set the timestamp in the future but to "future date - days to expire". This way the expiration job should have accurate value.

@jvillafanez did you test the current behavior on your PR and double check what "days" value you see in the notification ?

Reply:
history timestamp in the history table is supposed to be "$expireDate - $daysUntilExpiry" so that the notification can show "your password will expire in $daysUntilExpiry" notification.

now how to calculate "$daysUntilExpiry" here in a way that makes it work... should we use the existing rule ?
maybe `$daysUntilExpiry = time() - $date->getTimestamp()` ?

Reply:
this is the formula: `$changeTime = $date->getTimestamp() - $passwordExpiresAfterDaysRule`

Reply:
- [ ] regarding `\max` => don't insert dummy entry if the previous entry is more recent than the calculated `$changeTime`